src/host/process_launcher.h

```diff
@@ -12,6 +12,178 @@
 
 namespace asynchost
 {
+  struct ProcessPipe : public with_uv_handle<uv_pipe_t>
+  {
+  public:
+    ProcessPipe()
+    {
+      uv_handle.data = this;
+      uv_pipe_init(uv_default_loop(), &uv_handle, 0);
+    }
+    virtual ~ProcessPipe() = default;
+
+    uv_stream_t* stream()
+    {
+      return (uv_stream_t*)&uv_handle;
+    }
+
+  protected:
+    int pid = 0;
+  };
+
+  /**
+   * Read the output of a process line by line and print each one to our logs.
+   */
+  class ProcessReader : public ProcessPipe
+  {
+  public:
+    ProcessReader(std::string name) : name(name) {}
+
+    void start(pid_t pid)
+    {
+      this->pid = pid;
+
+      int rc = uv_read_start((uv_stream_t*)&uv_handle, on_alloc_cb, on_read_cb);
+      if (rc < 0)
+      {
+        LOG_FAIL_FMT("uv_read_start failed: {}", uv_strerror(rc));
+        close();
+      }
+    }
+
+  private:
+    static void on_alloc_cb(
+      uv_handle_t* handle, size_t suggested_size, uv_buf_t* buf)
+    {
+      static_cast<ProcessReader*>(handle->data)->on_alloc(suggested_size, buf);
+    }
+
+    static void on_read_cb(
+      uv_stream_t* handle, ssize_t nread, const uv_buf_t* buf)
+    {
+      static_cast<ProcessReader*>(handle->data)->on_read(nread, buf);
+    }
+
+    void on_alloc(size_t suggested_size, uv_buf_t* buf)
+    {
+      auto alloc_size = std::min<size_t>(suggested_size, 1024);
+      LOG_TRACE_FMT(
+        "Allocating {} bytes for reading from host process pid={}",
+        alloc_size,
+        pid);
+
+      buf->base = new char[suggested_size];
```

Comment:
That seems like it should be alloc_size?

Reply:
Yeah of course. I also noticed the TCP code uses 16KB, so I switched to that for consistency. I don't know if the value matters much tbh.